docs/plugins/development/creating-text-analysis-plugins.asciidoc

```diff
@@ -0,0 +1,129 @@
+[[creating-text-analysis-plugins]]
+=== Creating text analysis plugins
+
+Text analysis plugins provide {es} with custom Lucene analyzers, token filters, 
+character filters, and tokenizers. 
+
+[discrete]
+==== Plugin file structure
+
+Plugins are ZIP files composed of JAR files and two metadata files:
+
+* `stable-plugin-descriptor.properties` - a Java properties file that describes 
+the plugin. Refer to <<stable-plugin-descriptor.properties>>.
+* `named_components.json` - a JSON file mapping interfaces to key-value pairs 
+of component names and implementation classes.
+
+Note that only JAR files at the root of the plugin are added to the classpath
+for the plugin. If you need other resources, package them into a resources JAR.
+
+[discrete]
+==== The analysis plugin API
+
+Text analysis plugins can be written against the analysis plugin API. This API
```

Comment:
i wonder if it wouldn't read better if we just say something along the lines
```
Text analysis plugins are written against the analysis plugin API which consists of the following dependencies:
plugin-api
plugin-analysis-api
lucene-analysis-common
 
```

Reply:
Would you drop the descriptions for each the dependencies, for example "which contains the classes that an analysis plugin can implement"?

Reply:
- plugin-api - an api used by plugin developers to implement custom Elasticsearch plugins (stable)
- plugin-analysis-api - an api used by plugin developers to implement analysis plugins and integrate them into Elasticsearch
- lucene-analysis-common - a dependency of plugin-analysis-api contains core lucene analysis interfaces like Tokenizer, Analyzer, TokenStream